prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -4075,9 +4078,14 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         loop_head: BasicBlockIndex,
         array_base: vir::Expr,
     ) -> vir::Expr {
-        // this label is generated and inserted in encode_loop_invariant_exhale_stmts at the point
+        // this label is inserted in encode_loop_invariant_exhale_stmts at the point
         // where the other "preserve equality" assignments are made
-        let old_label = format!("loop_preserve_values_label_{:?}", loop_head);
+        use std::collections::hash_map::Entry::*;
+        let old_label = match self .array_loop_old_label.entry(loop_head) {
+            Occupied(lbl) => lbl.into_mut(),
+            Vacant(v) => v.insert(self.cfg_method.get_fresh_label_name()),
+        };
```

Comment:
Is this not just:
```suggestion
        let old_label = self.array_loop_old_label.entry(loop_head)
            .or_insert_with(|| self.cfg_method.get_fresh_label_name());
```
?

Reply:
that's what i first had as well, but it moves self into the closure :-/

Reply:
Oh, that… Oh well, we can leave this. Nitpick:
```suggestion
        use std::collections::hash_map::Entry::*;
        let old_label = match self.array_loop_old_label.entry(loop_head) {
            Occupied(lbl) => lbl.into_mut(),
            Vacant(v) => v.insert(self.cfg_method.get_fresh_label_name()),
        };
```